packages/mdc-dialog/component.ts

```diff
@@ -0,0 +1,186 @@
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+import {MDCComponent} from '@material/base/component';
+import {SpecificEventListener} from '@material/base/types';
+import {closest, matches} from '@material/dom/ponyfill';
+import {MDCRipple} from '@material/ripple/index';
+import * as FocusTrapLib from 'focus-trap';
+import {MDCDialogFoundation} from './foundation';
+import {FocusTrapFactory} from './types';
+import * as util from './util';
+
+const strings = MDCDialogFoundation.strings;
+
+export type MDCDialogFactory = (root: Element, foundation?: MDCDialogFoundation) => MDCDialog;
+
+export class MDCDialog extends MDCComponent<MDCDialogFoundation> {
+  get isOpen() {
+    return this.foundation_.isOpen();
+  }
+
+  get escapeKeyAction() {
+    return this.foundation_.getEscapeKeyAction();
+  }
+
+  set escapeKeyAction(action) {
+    this.foundation_.setEscapeKeyAction(action);
+  }
+
+  get scrimClickAction() {
+    return this.foundation_.getScrimClickAction();
+  }
+
+  set scrimClickAction(action) {
+    this.foundation_.setScrimClickAction(action);
+  }
+
+  get autoStackButtons() {
+    return this.foundation_.getAutoStackButtons();
+  }
+
+  set autoStackButtons(autoStack) {
+    this.foundation_.setAutoStackButtons(autoStack);
+  }
+
+  static attachTo(root: Element) {
+    return new MDCDialog(root);
+  }
+
+  private buttonRipples_!: MDCRipple[]; // assigned in initialize()
+  private buttons_!: HTMLElement[]; // assigned in initialize()
+  private container_!: HTMLElement; // assigned in initialize()
+  private content_!: HTMLElement | null; // assigned in initialize()
+  private defaultButton_!: HTMLElement | null; // assigned in initialize()
+  private initialFocusEl_!: HTMLElement | null; // assigned in initialize()
+
+  private focusTrapFactory_?: FocusTrapFactory; // assigned in initialize()
+  private focusTrap_!: FocusTrapLib.FocusTrap; // assigned in initialSyncWithDOM()
+
+  private handleInteraction_!: SpecificEventListener<'click'|'keydown'>; // assigned in initialSyncWithDOM()
+  private handleDocumentKeydown_!: SpecificEventListener<'keydown'>; // assigned in initialSyncWithDOM()
+  private handleLayout_!: EventListener; // assigned in initialSyncWithDOM()
+  private handleOpening_!: EventListener; // assigned in initialSyncWithDOM()
+  private handleClosing_!: () => void; // assigned in initialSyncWithDOM()
+
+  initialize(
+      focusTrapFactory?: FocusTrapFactory,
```

Comment:
The default assignment was taken off of this argument. Is it never relied upon?

Reply:
Not directly, no. `util.createFocusTrapInstance()` specifies the exact same default value, so it seemed pointless to duplicate that here.